Makefile

```diff
@@ -1,8 +1,11 @@
+CRATE_VERSION=$(shell head -4 Cargo.toml | grep ^version | awk '{print $$3}' | tr -d \")
+
 build_docker:
-	docker build --no-cache -t omidmr/gitversion docker
+	docker build --no-cache -t omidmr/gitversion -t omidmr/gitversion:$(CRATE_VERSION) docker
 
 push_docker:
-	docker push omidmr/gitversion
+	docker push omidmr/gitversion:latest
```

Comment:
I think the _latest_ tag doesn't exist at this stage or I'm wrong?!

Reply:
The way I was intending to go about it, if you've already run `make build_docker` and it has run successfully, then you should always have 2 resulting tags. `latest` and the `cargo pkgid` versions are simultaneously tagged.

I ran the make rule to confirm.

```
$ make build_docker
docker build --no-cache -t omidmr/gitversion -t omidmr/gitversion:0.1.0 docker
Sending build context to Docker daemon  2.048kB
Step 1/6 : FROM rust:1 as builder
[...]
Successfully tagged omidmr/gitversion:latest
Successfully tagged omidmr/gitversion:0.1.0
```

However, in the event that you've run this by accident, you could potentially update `latest` unintentionally.

What I mean is reversing the order of the `docker push` could serve as a gate for not accidentally pushing an older cached image. If attempting to push the pkgid tagged image fails, then latest won't be modified in the docker registry.

Thoughts?

Reply:
OK, yes, you are right and the way you have in your mind is fine :+1: